### PR TITLE
Frend n+1

### PR DESCRIPTION
**Pull request by mism-mism**

### CODE REVIEW

**Comment by mism-mism** [comment]
# 1 

一旦この点数
> {"success"=>2191, "redirect"=>614, "failure"=>1, "error"=>0, "exception"=>0}

File: webapp/go/app.go [hunk 1]
```diff
@@ -336,9 +336,14 @@ LIMIT 10`, user.ID)
...
 		commentsForMe = append(commentsForMe, c)
 	}
-	rows.Close()
 
-	rows, err = db.Query(`SELECT * FROM entries ORDER BY created_at DESC LIMIT 1000`)
+	rows, err = db.Query(`SELECT * FROM (
+    	(SELECT e.* FROM entries AS e JOIN relations AS r on e.user_id = r.another WHERE r.another = ? GROUP BY e.id ORDER BY e.id DESC LIMIT 20) 
+    	UNION 
+    	(SELECT e.* FROM entries AS e JOIN relations AS r on e.user_id = r.one WHERE r.one = ? GROUP BY e.id ORDER BY e.id DESC LIMIT 20)) AS ue 
+		GROUP BY ue.id ORDER BY ue.id DESC LIMIT 10`,
```
Comment (pinkumohikan) [inline webapp/go/app.go]: この場合はjoinじゃなくてsub queryで書いたほうが素直になりそう

```
select *
from entries
where
user_id in (友達)
order by id desc limit 10
```
的な

**Comment by mism-mism** [comment]
> {"success"=>4967, "redirect"=>1366, "failure"=>1, "error"=>0, "exception"=>0}

**Comment by mism-mism** [comment]
https://github.com/pinkumohikan/isucon-practice-20210724/issues/2#issuecomment-886002634
https://github.com/pinkumohikan/isucon-practice-20210724/issues/2#issuecomment-886002784

**Comment by mism-mism** [comment]
> {"success"=>4978, "redirect"=>1370, "failure"=>1, "error"=>0, "exception"=>0}

**Pull request merged**